File: src/main/python/leetcode_by_python/array/No118_Pascals_Triangle.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Created by HazzaCheng on 2019-11-05
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = []

        for i in range(numRows):
            row = []
            for j in range(i + 1):
                if j == 0 or j == i:
                    row.append(1)
                else:
                    row.append(res[-1][j] + res[-1][j - 1])
            res.append(row)
        return res

File: src/main/java/com/hazza/leetcode/array/No118PascalsTriangle.java
package com.hazza.leetcode.array;

import java.util.ArrayList;
import java.util.List;

/**
 * Created with IntelliJ IDEA.
 * Description: https://leetcode.com/problems/pascals-triangle/description/
 * User: HazzaCheng
 * Contact: hazzacheng@gmail.com
 * Date: 18-1-30
 * Time: 2:13 PM
 */
public class No118PascalsTriangle {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> row = new ArrayList<>();

        for (int i = 0; i < numRows; i++) {
            row.add(0, 1);
            for (int j = 1; j < i; j++)
                row.set(j, row.get(j) + row.get(j + 1));
            res.add(new ArrayList<>(row));
        }

        return res;
    }
}
